atlp-rwanda/hackers-ec-Fe
# #187419042 setup test locally with Vitest and CI

#### What does this PR do?
This PR introduces Vitest for unit testing, sets up a Continuous Integration (CI) pipeline with github action, adds various status badges to the README, and integrates CodeClimate for code quality analysis.

#### Description of Task to be completed?
- Set up Vitest for unit testing in the project.
- Implement a CI pipeline using GitHub Actions.
- Add coverage and test status badges to the README.
- Integrate CodeClimate and add its badge to the README.
- Add badges for all technologies used in the project to the README.

#### How should this be manually tested?
1. Clone the repository.
2. Check out to the branch `ch-setup-test-and-CI-187419042`.
3. Run `npm install` to install dependencies.
4. Run `npm run test` to execute tests using Vitest.
5. Verify that all tests pass and coverage is reported.

#### Any background context you want to provide?
Vitest has been chosen for its performance benefits and compatibility with the Vite build tool we are using.

#### What are the relevant pivotal tracker/Trello stories?
187419042
